qualtran/bloqs/rotations/hamming_weight_phasing.ipynb

```diff
@@ -0,0 +1,290 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "a9313ce9",
+   "metadata": {
+    "cq.autogen": "title_cell"
+   },
+   "source": [
+    "# Z Rotations via Hamming Weight Phasing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c3a3a9be",
+   "metadata": {
+    "cq.autogen": "top_imports"
+   },
+   "outputs": [],
+   "source": [
+    "from qualtran import Bloq, CompositeBloq, BloqBuilder, Signature, Register\n",
+    "from qualtran import QBit, QInt, QUInt, QAny\n",
+    "from qualtran.drawing import show_bloq, show_call_graph, show_counts_sigma\n",
+    "from typing import *\n",
+    "import numpy as np\n",
+    "import sympy\n",
+    "import cirq"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e9b47fcd",
+   "metadata": {
+    "cq.autogen": "HammingWeightPhasing.bloq_doc.md"
+   },
+   "source": [
+    "## `HammingWeightPhasing`\n",
+    "Applies $Z^{\\text{exponent}}$ to every qubit of an input register of size `bitsize`.\n",
+    "\n",
+    "Hamming weight phasing reduces the number of rotations to be synthesized from $n$ (where\n",
+    "$n=\\text{bitsize}$ is the size of the input register) to $\\log_2(n)$ via the following steps:\n",
+    "    1. Compute the hamming weight (HW) of the input register in using (at-most) $n-1$ ancilla\n",
+    "        and Toffolis in a newly allocated output register of size $\\log_2(n)$.\n",
+    "        $HW|x\\rangle -> |x\\rangle |\\text{HW}(x)\\rangle$\n",
+    "    2. Apply $Z^{2^{k}\\text{exponent}}$ to the k'th qubit of newly allocated hamming weight\n",
+    "         register.\n",
+    "    3. Uncompute the hamming weight register and ancillas allocated in Step-1 with 0 Toffoli\n",
+    "        cost.\n",
+    "\n",
+    "Overall, for an input register of size $n$, the procedure uses $n - \\alpha$ Toffoli's and\n",
+    "$n - \\alpha + \\log_2(n)$ ancilla to reduce $n$ rotation syntheses into $\\log_2(n)$  rotation\n",
+    "synthesis. Here $\\alpha = \\text{hamming\\_weight}(n)$.\n",
+    "\n",
+    "#### Parameters\n",
+    " - `bitsize`: Size of input register to apply `Z ** exponent` to.\n",
+    " - `exponent`: The exponent of `Z ** exponent` to be applied to each qubit in the input register.\n",
+    " - `eps`: Accuracy of synthesizing the Z rotations. \n",
+    "\n",
+    "#### Registers\n",
+    " - `A single THRU register of size `bitsize`.`:  \n",
+    "\n",
+    "#### References\n",
+    " - [Halving the cost of quantum addition](https://arxiv.org/abs/1709.06648). , Page-4\n"
+   ]
+  },
```

Comment:
This is taken from the docstring and displayed as markdown in the jupyter notebook. Some of the latex doesn't render very nicely. The "Registers" section also isn't formatted correctly in the docstring. It should be like the `Args` section, namely `reg_name: a brief description`. This one is missing the register name. 

I know you didn't write the docstring, but could you do a quick once-over on the notebook and fix some of these issues that only show up when it's displayed in the markdown/notebook format